private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -327,6 +329,46 @@ func NewProtoModulePinsForModulePins(modulePins ...ModulePin) []*modulev1alpha1.
 	return protoModulePins
 }
 
+// ModulePinResolver is a resolver for module pins.
+type ModulePinResolver interface {
```

Comment:
We should not be adding this to `bufmoduleref` - if anywhere, it would go in `bufmodule`, and probably just be another method on `ModuleResolver`.